lib/interferon/destinations/datadog.rb

```diff
@@ -270,40 +276,43 @@ def update_datadog_alert(alert, datadog_query, message, alert_options, existing_
         options: alert_options,
       }
 
-      if @dry_run
-        resp = @dog.validate_monitor(
-          alert['monitor_type'],
-          datadog_query,
-          monitor_options
-        )
-      elsif self.class.same_monitor_type(alert['monitor_type'], existing_alert['type'])
-        resp = @dog.update_monitor(
-          id,
-          datadog_query,
-          monitor_options
-        )
-
-        # Unmute existing alerts that exceed the max silenced time
-        # Datadog does not allow updates to silencing via the update_alert API call.
-        silenced = existing_alert['options']['silenced']
-        if !@max_mute_minutes.nil?
-          silenced = silenced.values.reject do |t|
-            t.nil? || t == '*' || t > Time.now.to_i + @max_mute_minutes * 60
-          end
-          @dog.unmute_monitor(id) if alert_options[:silenced].empty? && silenced.empty?
-        elsif alert_options[:silenced].empty? && !silenced.empty?
-          @dog.unmute_monitor(id)
-        end
-      else
-        # Need to recreate alert with new monitor type
-        resp = @dog.delete_monitor(id)
-        code = resp[0].to_i
-        unless code >= 300 || code == -1
-          resp = @dog.monitor(
+      resp = ''
+      retryable do
```

Comment:
If we wrap this entire block, we'd be retrying the update monitor call if silenced calls failed